File: AquivosJava/CPF.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

// Classe responsável por controlar os CPFs
public class CPF {
    // Atributo privado que guarda a lista de CPFs
    private List<String> fila;

    // Construtor que inicializa a lista
    public CPF() {
        this.fila = new ArrayList<>();
    }

    // Método para inserir CPF na fila
    public void inserirCPF(String cpf) {
        // Verifica se o CPF já existe
        if (fila.contains(cpf)) {
            System.out.println("CPF já cadastrado!");
        } else {
            fila.add(cpf); // Adiciona CPF à lista
            System.out.println("CPF inserido com sucesso!");
        }
    }

    // Método para listar CPFs (com senha)
    public void listarCPF(String senha) {
        if (senha.equals("redes2025")) {
            if (fila.isEmpty()) {
                System.out.println("Fila vazia!");
            } else {
                System.out.println("CPFs na fila:");
                for (String cpf : fila) {
                    System.out.println(cpf);
                }
            }
        } else {
            System.out.println("Senha incorreta!");
        }
    }

    // Método para remover CPF da fila
    public void removerCPF(String cpf) {
        if (fila.contains(cpf)) {
            fila.remove(cpf);
            System.out.println("CPF removido com sucesso!");
        } else {
            System.out.println("CPF não encontrado na fila!");
        }
    }

    // Getter para acessar a lista
    public List<String> getFila() {
        return fila;
    }

    // Setter
    public void setFila(List<String> fila) {
        this.fila = fila;
    }

    // Método principal com menu
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        CPF cpfManager = new CPF();
        int opcao;

        // Menu interativo usando switch
        do {
            System.out.println("\n--- MENU ---");
            System.out.println("1 - Inserir um CPF na fila");
            System.out.println("2 - Listar os CPFs da fila");
            System.out.println("3 - Remover um CPF da fila");
            System.out.println("4 - Sair");
            System.out.print("Escolha uma opção: ");
            opcao = sc.nextInt();
            sc.nextLine(); // Limpa buffer

            switch (opcao) {
                case 1:
                    System.out.print("Digite o CPF (apenas números): ");
                    String cpfInserir = sc.nextLine();
                    // Valida se contém apenas números
                    if (cpfInserir.matches("\\d+")) {
                        cpfManager.inserirCPF(cpfInserir);
                    } else {
                        System.out.println("CPF inválido! Digite apenas números.");
                    }
                    break;

                case 2:
                    System.out.print("Digite a senha: ");
                    String senha = sc.nextLine();
                    cpfManager.listarCPF(senha);
                    break;

                case 3:
                    System.out.print("Digite o CPF para remover: ");
                    String cpfRemover = sc.nextLine();
                    cpfManager.removerCPF(cpfRemover);
                    break;

                case 4:
                    System.out.println("Programa encerrado.");
                    break;

                default:
                    System.out.println("Opção inválida!");
            }
        } while (opcao != 4);

        sc.close();
    }
}

File: ArquivoPython/CPF.py
# Classe responsável por controlar os CPFs
class CPF:
    def __init__(self):
        # Inicializa a lista de CPFs como atributo privado
        self.__fila = []

    # Método para inserir CPF na fila
    def inserirCPF(self, cpf):
        if cpf in self.__fila:
            print("CPF já cadastrado!")
        else:
            self.__fila.append(cpf)
            print("CPF inserido com sucesso!")

    # Método para listar CPFs (com senha)
    def listarCPF(self, senha):
        if senha == "redes2025":
            if len(self.__fila) == 0:
                print("Fila vazia!")
            else:
                print("CPFs na fila:")
                for cpf in self.__fila:
                    print(cpf)
        else:
            print("Senha incorreta!")

    # Método para remover CPF da fila
    def removerCPF(self, cpf):
        if cpf in self.__fila:
            self.__fila.remove(cpf)
            print("CPF removido com sucesso!")
        else:
            print("CPF não encontrado na fila!")

    # Getter para acessar a lista
    def getFila(self):
        return self.__fila


# Função principal com menu interativo usando match-case
def main():
    cpfManager = CPF()

    while True:
        print("\n--- MENU ---")
        print("1 - Inserir um CPF na fila")
        print("2 - Listar os CPFs da fila")
        print("3 - Remover um CPF da fila")
        print("4 - Sair")

        try:
            opcao = int(input("Escolha uma opção: "))
        except ValueError:
            print("Opção inválida! Digite um número.")
            continue

        match opcao:
            case 1:
                cpfInserir = input("Digite o CPF (apenas números): ")
                # Valida se contém apenas números
                if cpfInserir.isdigit():
                    cpfManager.inserirCPF(cpfInserir)
                else:
                    print("CPF inválido! Digite apenas números.")

            case 2:
                senha = input("Digite a senha: ")
                cpfManager.listarCPF(senha)

            case 3:
                cpfRemover = input("Digite o CPF para remover: ")
                cpfManager.removerCPF(cpfRemover)

            case 4:
                print("Programa encerrado.")
                break

            case _:
                print("Opção inválida!")


# Executa o programa
if __name__ == "__main__":
    main()
